### ISSUE TITLE
Support Kitty Image Protocol / MIME Types / Rendering

### ISSUE THREAD

**Issue by zachlloyd** [warpdotdev/Warp#26]
https://sw.kovidgoyal.net/kitty/graphics-protocol/

**Comment by elviskahoro** [comment]
iTerm2 is able to display images (and animated gifs) within the terminal, see: https://iterm2.com/documentation-images.html.

It would be cool if this was supported within warp as well.

**Comment by elviskahoro** [comment]
### Describe the solution you'd like?
```
Support for rendering non-text based mime-types, e.g. svgs, pngs
```

### Is your feature request related to a problem? Please describe.

```
Graphics
```

### Additional context

```
Sourced from Discord: (general channel: 07/28/2021)
"given that the UI framework can render arbitrary object, and like jupyter you have both a) a client server architecture and b) the notion of "cells", do you think you'll ever implement rendering of non-text based mime-types, e.g. svgs, pngs, ... for interactive plots like plotly?"
```
https://discord.com/channels/851854972600451112/870145535993077780/885405266370256907

**Comment by elviskahoro** [comment]
Unexpected characters composition

From @BastienM via #615

### Describe the bug

```shell
When trying to write `find`, the term ends up drawing a phone icon. But is interpreted correctly.

Tested with `kitty 0.24.1` using the same font `Incosolata Nerd Mono` and I got the correct behavior.
```


### To Reproduce

Steps to reproduce:
1. Open a fresh term
2. Write `f` then `i`


### Expected behaviour

```shell
`f + i` should draw `fi`
```


### Screenshots

![term](https://imgur.com/U2JlYOC.png)

### Operating System

MacOS

### OS Version

12.1

### Additional context

```shell
Build `v0.2022.01.10.17.24.stable_02`
```

**Comment by klamonte** [comment]
Additional resources:

* I know you are aware of [notcurses](https://github.com/dankamongmen/notcurses).  Aim to survive the notcurses-demo, it is a very good test of performance, Unicode support, 24-bit RGB, memory pressure, and images (both kitty and sixel).

* If you choose sixel (perhaps from merging ayosec's graphics PR branch from alacritty), then you should also be aware of [chafa](https://hpjansson.org/chafa/) .  Among its encoders is an extremely fast high-quality sixel encoder.  notcurses and chafa are the fastest encoders I know of, and have different performance characters on the high end.  If you can keep up with both of them, then you will be in great shape.

* If you choose either sixel or iTerm2, then you should also test against [jexer](https://jexer.sourceforge.io/sixel.html) . When I wrote Jexer's image backend, only 3 terminals survived its lots-and-lots-of-small-images approach.  Now most terminals are working OK (though some still crash).  If you retrofit ayosec's graphics PR from alacritty, it should work fine with Jexer-type output, but will not work fine for notcurses output due to how it treats partially-damaged images.

On a different topic, just so I don't spam you with junk:

* Since you started from alacritty, you should be good with vttest now as they stopped crashing on it 5-ish years ago.  But their coverage of vttest/VT100/xterm is still far from complete.  If very good VT100/xterm coverage including VT52 submode is one of your goals, then you will need to go a bit past the Paul Flo Williams parser.  (Specifically it does have a state for the VT52 direct cursor address sequence, which isn't terminated like ECMA-48/ANSI X3.64 sequences are supposed to be.)

**Comment by elviskahoro** [comment]
@klamonte thanks for sharing!

**Comment by klamonte** [comment]
You're welcome.  I'm rooting for you! https://twitter.com/AutumnMeowMeow/status/1489320880471748613

**Comment by elviskahoro** [comment]
From elna via Discord: https://discord.com/channels/851854972600451112/870145535993077780/919330371647180832

> https://github.com/tauri-apps/wry - how cool would it be if we could display html etc in warp? terminal based notebooks a la jupyter with rendering things like plotly plots would be possible

**Comment by klamonte** [comment]
Jexer and notcurses both mix images and text, but with very different internal models. I can't speak much to notcurses' model, but I did outline jexer's here: https://gitlab.com/klamonte/jexer/-/wikis/high-level-design#text-cells

So far it has proven capable of doing almost everything we expect GUI things to do, without asking for significant investment in features from the terminal itself. Performance with animations has been a challenge though, and I do not know if that is from my use of Java or the model itself.

**Comment by elviskahoro** [comment]
https://gitlab.com/gnachman/iterm2/-/issues/6436

**Comment by japanese-goblinn** [comment]
Any updates on this? Really cool feature 😅

**Comment by elviskahoro** [comment]
@japanese-goblinn No updates. I'll share info in this github issue when there is. My guess is that this wouldn't be soon though.

**Comment by elviskahoro** [comment]
@japanese-goblinn No updates. I'll share info in this github issue when there is. My guess is that this wouldn't be soon though.

**Comment by japanese-goblinn** [comment]
@elviskahoro okey, thanks!

**Comment by elviskahoro** [comment]
From @wellcomez via #1754

Protocol
iTerm2 extends the xterm protocol with a set of proprietary escape sequences. In general, the pattern is:

ESC ] 1337 ; key = value ^G
Whitespace is shown here for ease of reading: in practice, no spaces should be used.

For file transfer and inline images, the code is:

ESC ] 1337 ; File = [arguments] : base-64 encoded file contents ^G
The arguments are formatted as key=value with a semicolon between each key-value pair. They are described below:

**Comment by mikeal** [comment]
any updates on this? love this project but can't switch until this works :)

**Comment by warpdotdev-dx** [comment]
No updates. This wouldn't be for awhile it's very high engineering lift!

**Comment by amjad1233** [comment]
Is this one why emoji's are not rendering or should I open a new issue for it. Bascially Emojis are coming as wierd character please see bleow screenshot, where emoji should render as Aussie flag.
![CleanShot 2022-11-28 at 08 57 25](https://user-images.githubusercontent.com/661146/204164393-eab9e708-0fc3-4d82-ae7c-8c673ad7fffb.png)

**Comment by p-i-c-o** [comment]
@warpdotdev-dx any updates??

**Comment by dannyneira** [comment]
@p-i-c-o As far as I'm aware, there is some work being done on gif support, but nothing for kitty image support yet.

**Comment by scottbea** [comment]
It's honestly hard to take Warp serious if it does not offer the basics like rendering inline images, animated gifs, and images. I'd expect rendering to support pixel-based and character-based rendering both, with scaling, opacity control, caching for animations, etc.  Even more, I'd expect pixel level R/W access to the terminal window (with plenty of security controls which makes this non-trivial).  Also, plan now on how to support voxel rendering in the terminal because 3D matters.

There are some great comments above about the specifics around ensuring the terminal can perform very well with many hundreds of embedded / inline images.  Yes, let's make it very fast please.

Think what we did with the C64 in the 80s...  Let's see some radical stuff here!

**Comment by RobbieMcKinstry** [comment]
Ratatui, the Rust TUI framework, added experimental Sixel support [as described here](https://blog.orhun.dev/ratatui-0-23-0/). I'm salivating at how great this would look in Warp! :D

**Comment by t0msk** [comment]
+1 I wish this feature

**Comment by sneazy-ibo** [comment]
any updates on this for warp?

**Comment by t0msk** [comment]
hmm its 3 years old feature request :( I mean modern terminal should support graphics rendering

**Comment by sneazy-ibo** [comment]
fr I got so many cli tools that use images, not having them visible on warp is really a pain

**Comment by Cryolitia** [comment]
need it, too

**Comment by ALEZ-DEV** [comment]
I'm really surprised that warp does not already support this

**Comment by Ast3risk-ops** [comment]
> I'm really surprised that warp does not already support this

It should be a feature for a terminal that brands itself as "your terminal reimagined."

**Comment by MilkTicc** [comment]
Any update on this? Really need it

**Comment by ZhangHanDong** [comment]
any updates on this for warp?

**Comment by neuromuse** [comment]
I love Warp but this / Sixel support is a requirement for it to be my daily driver. Can we please get some kind of developer response here? This request has been sitting here for 3 years.

**Comment by danischeer** [comment]
yeah, would love to use warp but no image support, no chance. When i do Frontend Development i'm constantly in Yazi for blaszingly fast codebase searching with previews, will switch to warp the day image preview is available, without it its just not worth the tradeoff.

**Comment by mkubasz** [comment]
One more vote for this feature

**Comment by kitty-eu-org** [comment]
+1

**Comment by BingCoke** [comment]
+1

**Comment by dabing1022** [comment]
+1

**Comment by Zedelghem** [comment]
+1, crucial for web browsing from CLI

**Comment by akiidjk** [comment]
+ 1

**Comment by ddooley2** [comment]
+1

**Comment by LuisFellp** [comment]
+1

**Comment by ALEZ-DEV** [comment]
+1

**Comment by Ast3risk-ops** [comment]
We get it, thanks.

**Comment by Ghussy** [comment]
+1 I dont want to use kitty, pls

**Comment by odebroqueville** [comment]
+1

**Comment by tanjuntao** [comment]
One more request for this feature!!! Really need it.

**Comment by tariqwest** [comment]
+1

**Comment by donCESAR12345** [comment]
+1

**Comment by tkjung92** [comment]
+1

**Comment by RWayne93** [comment]
+1 crazy this has been open for so long.

**Comment by mbradyneeley** [comment]
+1

**Comment by HelpHelpLe** [comment]
My desktop environment is GNOME 47.1 with Wayland enabled. Following the instructions at https://yazi-rs.github.io/docs/image-preview, I installed ueberzugpp, and I am currently able to preview images in yazi within Warp.
![Screenshot From 2024-11-11 15-57-25](https://github.com/user-attachments/assets/a361bd4c-5594-4db6-a88a-377606feec06)

**Comment by leaderlessllc** [comment]
+1

**Comment by fengjiawang** [comment]
+1

**Comment by daUnknownCoder** [comment]
> My desktop environment is GNOME 47.1 with Wayland enabled. Following the instructions at [yazi-rs.github.io/docs/image-preview](https://yazi-rs.github.io/docs/image-preview), I installed ueberzugpp, and I am currently able to preview images in yazi within Warp.

yes that works for me too, but the image placement is always quirky, and iirc, kovid goyal shared the kitty image protocol so that other terminals can also use it, it would be cool to see that bcz kitty's image implementation is v.good

**Comment by kvodenicharov** [comment]
+1

**Comment by rustnomicon** [comment]
+1

**Comment by h-mayorquin** [comment]
+ 1

**Comment by bearcatsandor** [comment]
This would be especially great for neovim users. A popular finder/explorer called Snacks.vim can use the kitty protocol to view image previews in neovim.

**Comment by daUnknownCoder** [comment]
> This would be especially great for neovim users. A popular finder/explorer called Snacks.vim can use the kitty protocol to view image previews in neovim. 

Yeah since ligatures have come already, image support will be be trivial

**Comment by BeginningGradeMaker** [comment]
> This would be especially great for neovim users. A popular finder/explorer called Snacks.vim can use the kitty protocol to view image previews in neovim.

Just wondering: are you guys comfortable using lsp without undercurl? I posted this issue but seems like nobody cares about this feature, which is builtin in many terminals.

**Comment by filiph** [comment]
I'm surprised nobody has linked yet to @katef's threads about graphics in terminals from about 5 years ago. They opened my eyes to what's possible on the terminal, and — more importantly — how useful that can be.

- https://x.com/thingskatedid/status/1316075850580652032
- https://x.com/thingskatedid/status/1386077306381242371

![Image](https://github.com/user-attachments/assets/d5980161-3803-4ffb-b1f2-4ff0dfb71f83)

![Image](https://github.com/user-attachments/assets/76ddd6e0-8a55-421a-bed8-6772bf00a173)

![Image](https://github.com/user-attachments/assets/2c123a0f-10b3-4b44-846b-fae46791c0ed)

These are all rendered via kitty (`isvg`, for example, is just an alias for `rsvg-convert | kitty icat --align=left`.

**Comment by chiragkrishna** [comment]
+1

**Comment by Olaison** [comment]
+1

**Comment by qq565425677** [comment]
+1

**Comment by aztlan2k** [comment]
So I noticed the latest release notes list the following:

> 2025.03.05 (v0.2025.03.05.08.02)
> New features
> 
> * iTerm Image Protocol is now supported on Mac and Linux!

Does this mean this is now at least partially suppoorted?

**Comment by bearcatsandor** [comment]
I guess l don't understand why the iterm protocol was used.  The  kitty protocol is superior in a number of ways, including GPU utilization, and this is a terminal that prides itself on that.

**Comment by Ast3risk-ops** [comment]
Thing is more stuff supports the kitty protocol than the iterm2 protocol.

**Comment by daUnknownCoder** [comment]
yep, kitty protocol is much more faster too if not the best, idk why this decision was made (maybe related to Warp being a Mac terminal utility first, and iTerm is the go to terminal application for Macs)

**Comment by juanlopez4691** [comment]
Images display has some issues with TUI apps, e.g. [ranger file manager](https://github.com/ranger/ranger). Last displayed images get stuck on display:

<img width="1390" alt="Image" src="https://github.com/user-attachments/assets/577248ad-7a95-4c2b-a519-7cd56a1a253d" />

To reproduce the issue, start ranger, preview an image, quit ranger, start it again in the same Warp tab.

**Comment by qq565425677** [comment]
> Images display has some issues with TUI apps, e.g. [ranger file manager](https://github.com/ranger/ranger). Last displayed images get stuck on display:
> 
> <img alt="Image" width="1390" src="https://private-user-images.githubusercontent.com/13467494/420629908-577248ad-7a95-4c2b-a519-7cd56a1a253d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE0ODk5MTIsIm5iZiI6MTc0MTQ4OTYxMiwicGF0aCI6Ii8xMzQ2NzQ5NC80MjA2Mjk5MDgtNTc3MjQ4YWQtN2E5NS00YzJiLWE1MTktN2NkNTZhMWEyNTNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDAzMDY1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwOTU2ZjhlOWY4NWVlY2E2YWU4ZGIwMTkzMWI5ZDNjZDRmZDI3OTQzYjI1OGMwNjJkZjkwZGI0OTA4ZDNlOGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qSOxU1uvnnXe5K0-OC3dJNWAnffp717vSjsvmGhd7cY">
> To reproduce the issue, start ranger, preview an image, quit ranger, start it again in the same Warp tab.
> 
> This doesn't happen on other terminals I tried (Ghostty, iTerm2).

+1. I have the same issues when using [yazi](https://github.com/sxyazi/yazi), a similar TUI app.

**Comment by juanlopez4691** [comment]
> +1. I have the same issues when using yazi, a similar TUI app.

Yes, I tríes with yazi and same issue happens. Also with superfile.

**Comment by juanlopez4691** [comment]
Funny (and shocking) new discovery: last image previewed in a Wrap terminal tab persist when opening other TUI apps, like lazygit:

<img width="1202" alt="Image" src="https://github.com/user-attachments/assets/2be48325-08ff-4f24-b0ac-2e1a1b8e42dc" />

**Comment by filiph** [comment]
FWIW, I think it would be helpful to scope this feature and maybe split it into at least two issues.

A) One is the relatively straighforward requirement to be able to show an image as output, e.g. `icat my_graph.png` shows a single image as program output, then continues to behave as a regular terminal.

B) The other would be a more complete implementation of graphical output on the command line, allowing things like TUI apps with images in them, animated images, videos, etc.

I, for one, would be completely satisfied with a good implementation of (A). Which, looking at https://github.com/warpdotdev/Warp/issues/26#issuecomment-2708309873, is maybe already finished?

I don't really care that much about (B), and I think there are more important and "core" things for the Warp engineering team to work on. Again, just my opinion, but I wanted to voice it lest someone at Warp assumes everyone on this thread roots for (B).

**Comment by juanlopez4691** [comment]
Well, perhaps I am mistaken but IMHO (B) is a symptom of a buggy implementation of (A).

**Comment by hiboy51** [comment]
https://github.com/sxyazi/yazi/issues/2511#issuecomment-2745110173

+1
please fix it #

**Comment by juanlopez4691** [comment]
+1
pleasy please fix it #

**Comment by SohamP2812** [comment]
Hey everyone! Warp now has support for the Kitty Image Protocol (and the iTerm protocol as mentioned earlier in the thread)! 

The bug with images persisting on screen when opening other TUI apps has been fixed (you may need to update to see the change). 

The bug specific to `yazi` where images are not being cleared has a fix and will be in the next update. We are also working to have `yazi` and other tools natively support images in Warp without having to set any environment vars.

We also currently do not support [unicode placeholders](https://sw.kovidgoyal.net/kitty/graphics-protocol/#unicode-placeholders) for Kitty and have an open issue to track this https://github.com/warpdotdev/Warp/issues/6210, so feel free to interact there if you find tools you use need that feature.

Hope y'all enjoy images in Warp finally being supported!

**Comment by h-mayorquin** [comment]
Thanks a bunch for the update @SohamP2812!

**Comment by hiboy51** [comment]
really appreciate it @SohamP2812

**Comment by idoa01** [comment]
I'm on warp `v0.2025.03.26.08.10.stable_02` and I can't see images at all in `yazi` (all I see is a bunch of unicode(?) question marks inside boxes). Is there anything I need to do to see the support of images in the terminal?

**Comment by ubuntudroid** [comment]
@idoa01 They are already on it: https://github.com/sxyazi/yazi/issues/2511 🚀

**Comment by csells** [comment]
> [@idoa01](https://github.com/idoa01?rgh-link-date=2025-03-29T16%3A26%3A42.000Z) They are already on it: [sxyazi/yazi#2511](https://github.com/sxyazi/yazi/issues/2511) 🚀

That issued is closed as not planned. I'm likewise on v0.2025.03.26.08.10.stable_02 and only see unicode squiggles for images in yazi.

**Comment by csells** [comment]
I have no idea how to see the contents of an image file in the Warp terminal and neither does Warp:

<img width="1012" alt="Image" src="https://github.com/user-attachments/assets/062753f7-e187-4513-8073-6cd9294e7a7f" />

**Comment by Ast3risk-ops** [comment]
> I have no idea how to see the contents of an image file in the Warp terminal and neither does Warp:
> <img alt="Image" width="1012" src="https://private-user-images.githubusercontent.com/2568253/429627431-062753f7-e187-4513-8073-6cd9294e7a7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM2MjUwOTgsIm5iZiI6MTc0MzYyNDc5OCwicGF0aCI6Ii8yNTY4MjUzLzQyOTYyNzQzMS0wNjI3NTNmNy1lMTg3LTQ1MTMtODA3My02Y2Q5Mjk0ZTdhN2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MDJUMjAxMzE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTMwMjA2ZDQzMmJhM2Q1MTg3NDAxYTE2NGFkMjRjMGY3ZDhkYTI3ZDAwM2ExZjE2M2M0YTFmZjNiMjhmNDQyYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.M2P5tycIfi7DS-g81hzHn914N9sIzrV02jI0ZEZwSxQ">

Use a tool like `ranger` or `termvisage`.

**Comment by bearcatsandor** [comment]
> I have no idea how to see the contents of an image file in the Warp terminal and neither does Warp:
> 
> <img width="1012" alt="Image" src="https://github.com/user-attachments/assets/062753f7-e187-4513-8073-6cd9294e7a7f" />

If you want to use the Kitty Image Protocol directly try:

kitty +kitten icat cover.jpg
or
icat cover.jpg

replacing "cover.jpg" with your image of choice. 'icat' is often an alias for the 1st command.

**Comment by SohamP2812** [comment]
@csells `termvisage` specifically checks for the terminal name to see if it supports Kitty (it uses characters as fallback to display images). We will be opening issues/PRs with tools soon to add Warp to the hard code check for support!

**Comment by csells** [comment]
> [@csells](https://github.com/csells?rgh-link-date=2025-04-03T18%3A57%3A52.000Z) `termvisage` specifically checks for the terminal name to see if it supports Kitty (it uses characters as fallback to display images). We will be opening issues/PRs with tools soon to add Warp to the hard code check for support!

Great! In the meantime, how do I see images in my Warp terminal today?

**Comment by SohamP2812** [comment]
> > [@csells](https://github.com/csells?rgh-link-date=2025-04-03T18%3A57%3A52.000Z) `termvisage` specifically checks for the terminal name to see if it supports Kitty (it uses characters as fallback to display images). We will be opening issues/PRs with tools soon to add Warp to the hard code check for support!
> 
> Great! In the meantime, how do I see images in my Warp terminal today?

Using [icat](https://sw.kovidgoyal.net/kitty/kittens/icat/) is my go to until we get other tools to support us with Kitty.

**Comment by j4james** [comment]
FYI, most image tools will let you force the image format via the command line if they don't recognise the terminal as one that supports the Kitty image protocol. Some examples:
```
timg -p kitty filename.png
termvisage -S kitty --force-style filename.png
chafa -f kitty filename.png
```

Although Warp's implementation of the protocol isn't yet good enough to support a lot of applications, so there's no guarantee that'll work. In particular, I think `chafa` depends on chunked encoding which Warp doesn't support.

**Comment by SohamP2812** [comment]
@j4james Good callout on the CLI args!

As for `chafa` - the chunk decoding issue has been fixed in the latest update (`v0.2025.04.02.08.11.stable_02`), so just need to update to see the fix.

Warp's implementation of the protocol should support most tools currently (beyond having to coerce them to use kitty with cli args or env vars). The primary things it is missing are unicode placeholders and animations.

**Comment by SohamP2812** [comment]
Closing this issue since the protocol support is released. Will be working to get other tools to use Kitty for Warp by default.

Feel free to open up new issues if you run into problems!

**Comment by csells** [comment]
> termvisage -S kitty --force-style filename.png

this works great. thank you!

**Comment by AnonymouX47** [comment]
@SohamP2812, thank you for your work.

I'm yet to test these features on Warp but as regards iTerm2's inline image protocol, I thought the following would be worthy of note:

See [this comment](https://github.com/jerch/xterm-addon-image/issues/58#issuecomment-1979939323) and the response (from iTerm2's author), especially the first sentence, in the comment following it. I don't know if this is implemented yet but if it isn't, it'd be good to have.

Also, see [this comment](https://github.com/jerch/xterm-addon-image/issues/57#issuecomment-1979734859) and the two following it.

I should be able to give some feedback when I get to test things out.

Thank you.

**Comment by dhruvkb** [comment]
I don't know what exactly the issue is but using `xterm-query` to send a Kitty image protocol query to warp raises an error `IO(Os { code: 22, kind: InvalidInput, message: "Invalid argument" })` and the response (i.e. `Gi=31;OK`) is actually entered into the terminal in the next input line.

<img width="1136" alt="Image" src="https://github.com/user-attachments/assets/a405c649-0f2f-4e8a-86b3-ddc4730ef462" />

Fwiw, this app works perfectly on WezTerm and Ghostty so I'm inclined to believe it's something about Warp's handling of [query responses](https://sw.kovidgoyal.net/kitty/graphics-protocol/#querying-support-and-available-transmission-mediums) specifically.

**Comment by SohamP2812** [comment]
Hmm do you mind providing more detail @dhruvkb

When I run `cargo run --example kitty` on https://github.com/Canop/xterm-query, it outputs "Kitty image protocol IS supported".

Also please make new issues from here on out whenever people find anything!

**Comment by dhruvkb** [comment]
@SohamP2812 I've added more details in this new issue #6284.

**Comment by Ast3risk-ops** [comment]
Just lock the issue at this point.
